File: linearSearch/linearSearch.js
const arrayToSearchThrough = [];
for (let i = 1; i <= 1000; i++) {
    arrayToSearchThrough.push(i);
}

exports.linearSearch = function(valueToFind, arrayToSearchThrough) {
    // your code here
    for (let i = 0; i < arrayToSearchThrough.length; i++) {
        if (arrayToSearchThrough[i] === valueToFind){
            return i
        }
    }

    return undefined
};

exports.linearSearchGlobal = function(valueToFind, arrayToSearchThrough) {
    // your code here
    indexArr = []
    for (let i = 0; i < arrayToSearchThrough.length; i++) {
        if (arrayToSearchThrough[i] === valueToFind){
            indexArr.push(i)
        }
    }

    return indexArr
};

File: linear_search_PY/linear_search.py
array_to_search_through = []
for number in range(1, 1001):
    array_to_search_through.append(number)

def linear_search(value_to_find, array_to_search_through):
    for idx in range(len(array_to_search_through)):
        if array_to_search_through[idx] == value_to_find:
            return idx
    
    return None 

def linear_search_global(value_to_find, array_to_search_through):
    answerArr = []
    for idx in range(len(array_to_search_through)):
        if array_to_search_through[idx] == value_to_find:
            answerArr.append(idx)

    return answerArr